editor/inspector/assets/material.js

```diff
@@ -4,6 +4,57 @@ const { materialTechniquePolyfill } = require('../utils/material');
 const { setDisabled, setReadonly, setHidden, loopSetAssetDumpDataReadonly } = require('../utils/prop');
 const { join, sep, normalize } = require('path');
 
+/**
+ * 
+ * @param {{name: string; uuid: string}[]} sortedEffects 
+ * @returns html template
+ */
+function renderGroupEffectOptions(sortedEffects) {
+    const groupNames = new Set();
+    const groupNameRE = /^(\w+)\//;
+    const effectDirRE = /^effects\//i;
+
+    let htmlTemplate = '';
+    let currGroup = '';
+
+    for (const effect of sortedEffects) {
+        const groupName = groupNameRE.exec(effect.name)?.[1] ?? '';
+        let optionLabel = effect.name;
+
+        // complete prev group
+        if (currGroup !== '' && currGroup !== groupName) {
+            htmlTemplate += '</optgroup>';
+        }
+
+        if (groupName !== '') {
+            if (!groupNames.has(groupName)) {
+                groupNames.add(groupName);
+
+                currGroup = groupName;
+
+                htmlTemplate += `<optgroup label="${groupName}">`;
+            }
+
+            // for option label, remove group name if matched
+            optionLabel = optionLabel.replace(groupNameRE, '');
```

Comment:
Is this path not displayed correctly? What will be displayed if I have files outside the "effect" folder?

Reply:
`internal/effects/a` => `effects/a`
`internal/effects/a/b` => `effects/a/b`
`internal/a` => `a`

Reply:
a !== a?
Isn't that ambiguous?